Added more outputs

ilyamandel opened the pull request:
- Added several outputs to provide more information and aid code testing: SEMI_MAJOR_AXIS_POST_STAGE_1_CE (for the 2-stage CE); BINDING_ENERGY_CONVECTIVE_ENVELOPE, CONVECTIVE_ENV_MASS, LAMBDA_CONVECTIVE_ENVELOPE (convective envelope binding energy,  mass, and lambda, respectively, Picker prescription)
- Ensure that the MANDEL main sequence core mass prescription never provides a core mass exceeding the total mass
- Corrected several erroneous comments, cleaned up a few function definitions
- Updated and corrected documentation

Commit:
Temporary commit

Commit:
Merge branch 'dev' into Outputs

Commit:
Documentation update

Commit:
Typo fix

jeffriley reviewed (approved):
Thanks @ilyamandel  - looks good.  Just one suggested change:

On line 293 of `BaseStar.cpp`, use `std::ignore` (https://en.cppreference.com/w/cpp/utility/tuple/ignore)

I'll approve now so you can merge after making the change.

EDIT: also, looks like a typo in Options.h (behavious -> behaviour(s))

EDIT: wait - you didn't change Options.h - why are we picking up a spelling mistake there now?

Commit:
Implementing Jeff's suggested change

Commit:
Typo fix

ilyamandel commented:
Thank you for the std::ignore suggestion; implemented.  I think the behavious typo was actually from the change you (@jeffriley ) just made, and which I rolled in via "git merge dev" after I already started on my PR, so it shows up in the changes relative to when I originally branched off dev; anyway, fixed that too, so will merge now.
